lib/wasix/src/syscalls/wasix/thread_spawn.rs

```diff
@@ -89,14 +89,33 @@ pub(crate) fn thread_spawn_internal<M: MemorySize>(
     let thread_id: Tid = thread_handle.id().into();
     Span::current().record("tid", thread_id);
 
+    // Spawn the thread
+    thread_spawn_internal_phase2::<M>(ctx, thread_handle, layout, start_ptr_offset, None)?;
+
+    // Success
+    Ok(thread_id)
+}
+
+pub fn thread_spawn_internal_phase2<M: MemorySize>(
```

Comment:
Comment: what are the phases, why are they needed?

Reply:
renamed to make it easier to understand